ibis/backends/bigquery/__init__.py

```diff
@@ -275,49 +288,75 @@ def drop_schema(
         force: bool = False,
         cascade: bool = False,
     ) -> None:
-        drop_stmt = "DROP SCHEMA"
-        if force:
-            drop_stmt += " IF EXISTS"
+        """Drop a BigQuery dataset."""
+        stmt = sg.exp.Drop(
+            kind="SCHEMA",
+            this=sg.table(name, db=database),
+            exists=force,
+            cascade=cascade,
+        )
 
-        drop_stmt += " "
-        drop_stmt += ".".join(filter(None, [database, name]))
-        drop_stmt += " CASCADE" if cascade else " RESTRICT"
-        self.raw_sql(drop_stmt)
+        self.raw_sql(stmt.sql(self.name))
 
-    def table(self, name: str, database: str | None = None) -> ir.TableExpr:
+    def table(
+        self, name: str, schema: str | None = None, database: str | None = None
```

Comment:
I think we should flip the order of `schema` and `database` -- also, if you use the pattern in `list_tables` we can get this in to 7.1.  I think this can be done in a non-breaking way, something like:

```python
        if database is not None and schema is None:
            util.warn_deprecated(
                "database",
                instead=(
                    f"{self.name} cannot list tables only using `database` specifier. "
                    "Include a `schema` argument."
                ),
                as_of="7.1",
                removed_in="8.0",
            )
            database = sg.parse_one(database, into=sg.exp.Table, read=self.name).sql(
                dialect=self.name
            )
        elif database is None and schema is not None:
            database = sg.parse_one(schema, into=sg.exp.Table, read=self.name).sql(
                dialect=self.name
            )
        else:
            database = sg.table(schema, db=database).sql(dialect=self.name) or None

        project, dataset = self._parse_project_and_dataset(database)
```

Reply:
Good points! I will take a look at this tomorrow and hopefully we can get it in for 7.1

Reply:
Done!